source/extensions/filters/http/rate_limit_quota/filter.cc

```diff
@@ -122,34 +124,37 @@ void RateLimitQuotaFilter::createNewBucket(const BucketId& bucket_id,
   // Create new bucket and store it into quota cache.
   std::unique_ptr<Bucket> new_bucket = std::make_unique<Bucket>();
 
-  // The first matched request doesn't have quota assignment from the RLQS server yet, so the
-  // action is performed based on pre-configured strategy from no assignment behavior config.
+  // The first matched request doesn't have quota assignment from the RLQS
+  // server yet, so the action is performed based on pre-configured strategy
+  // from no assignment behavior config.
   auto mutable_rate_limit_strategy =
-      new_bucket->bucket_action.mutable_quota_assignment_action()->mutable_rate_limit_strategy();
+      new_bucket->default_action.mutable_quota_assignment_action()->mutable_rate_limit_strategy();
   if (match_action.bucketSettings().has_no_assignment_behavior()) {
     *mutable_rate_limit_strategy =
         match_action.bucketSettings().no_assignment_behavior().fallback_rate_limit();
   } else {
-    // When `no_assignment_behavior` is not configured, default blanket rule is set to ALLOW_ALL.
-    // (i.e., fail-open).
-    mutable_rate_limit_strategy->set_blanket_rule(envoy::type::v3::RateLimitStrategy::ALLOW_ALL);
+    // When `no_assignment_behavior` is not configured, default blanket rule is
+    // set to ALLOW_ALL. (i.e., fail-open).
+    mutable_rate_limit_strategy->set_blanket_rule(RateLimitStrategy::ALLOW_ALL);
   }
 
   // Set up the bucket id.
   new_bucket->bucket_id = bucket_id;
-  // Set up the first time assignment time.
-  new_bucket->first_assignment_time = time_source_.monotonicTime();
+  // Mark the assignment time.
+  auto now = time_source_.monotonicTime();
+  new_bucket->first_assignment_time = now;
+  new_bucket->current_assignment_time = now;
```

Comment:
From quick scan, IIUC. this update is the key. 

I am wondering if other changes (like default action changes) are needed for the test update?

Reply:
The `default_action` field is required to differentiate from the default-action set by the `no_assignment_behavior` field.
The `cached_action` is set by RLQS responses as they give assignments for buckets.

These are notably different in that `cached_action` has a TTL set by an RLQS response, whereas the `default_action` does not ever expire, and must always be available to handle traffic before an assignment is given or after the latest assignment has been wiped by expiration.

We could potentially remove the actual `default_action` field from the Bucket and rely on dynamically reading from the `no_assignment_behavior` field of the `match_action`, but this change more closely matches existing code that only reads from `no_assignment_behavior` during `createNewBucket`.

Reply:
Added comments for clarity.